doc/src/site/sphinx/ml/sw_target_encoder.rst

```diff
@@ -120,6 +120,7 @@ Target Encoder in Sparkling Water is implemented as a regular estimator and thus
             val gbm = new H2OGBM()
                 .setFeaturesCols(targetEncoder.getOutputCols())
                 .setLabelCol("CAPSULE")
+                .setProblemType("Binomial")
```

Comment:
do we use "problem type" term other places in SW?

Reply:
No, just here. Other places use H2O-3 terms like `family` and `distribution`. Can you think of a better name for this field?

Reply:
I think it is strange to set both problem type and distribution/family on the H2OGBM object,
in other places you have

```

new H2OGBM()
   .setSeed(42)
   .setDistribution("bernoulli")
```

adding `setProblem` there with no connection to the distribution parameters is in my opinion not ideal.

We could 
 - drop the problem type parameter completely and let them set the distribution parameter
 - keep the problem type but make it optional and derive it automatically from the distribution parameter if it is passed
 
 @sebhrusen thoughts?

Reply:
from what I understand, `problemType` is useful only to decide if the response column should be converted to categorical, right? SW's approach being different from H2O-3 on this aspect, being more declarative.

Also by the way, `binomial` and `multinomial` usually refer to distributions, so using them as `problemType` values may confuse some. I would suggest to use `binary` and `multiclass` instead for those classification types.

This being said, I'd vote for your 2nd suggestion @michalkurka :
> keep the problem type but make it optional and derive it automatically from the distribution parameter if it is passed

and I don't see any better alternative.